### DISCUSSION TITLE
Service Level Agreement

### DISCUSSION THREAD

**Discussion by Jumaodhiss** [socketio/socket.io#4272] [Q&A] [answered]
I am trying to find out if there are documented service level agreement for delivering events and messages through socket io. Kindly help

**Comment by darrachequesne** [comment, marked as answer]
Hi! The messages are delivered in an at-most-once fashion.

That being said, additional guarantees could totally be implemented on top of Socket.IO.

See also: https://github.com/socketio/socket.io/issues/3649
